packages/medusa/src/api/routes/admin/inventory-items/create-inventory-item.ts

```diff
@@ -78,54 +77,106 @@ import { validator } from "../../../../utils/validator"
  */
 
 export default async (req, res) => {
-  const validated = await validator(AdminPostInventoryItemsReq, req.body)
-  const { variant_id, ...input } = validated
+  const { variant_id, ...inventoryItemInput } = req.validatedBody
 
-  const inventoryService: IInventoryService =
-    req.scope.resolve("inventoryService")
+  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
   const productVariantInventoryService: ProductVariantInventoryService =
     req.scope.resolve("productVariantInventoryService")
-  const productVariantService: ProductVariantService = req.scope.resolve(
-    "productVariantService"
-  )
-
-  let inventoryItems = await productVariantInventoryService.listByVariant(
-    variant_id
-  )
-
-  // TODO: this is a temporary fix to prevent duplicate inventory items since we don't support this functionality yet
-  if (inventoryItems.length) {
-    throw new MedusaError(
-      MedusaError.Types.NOT_ALLOWED,
-      "Inventory Item already exists for this variant"
-    )
-  }
 
-  const manager: EntityManager = req.scope.resolve("manager")
+  const createInventoryItemWorkflow = createInventoryItems(req.scope)
 
-  await manager.transaction(async (transactionManager) => {
-    await createInventoryItemTransaction(
+  if (!featureFlagRouter.isFeatureEnabled(ManyToManyInventoryFeatureFlag.key)) {
+    if (!variant_id) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_ALLOWED,
+        "variant_id is required"
+      )
+    }
+
+    createInventoryItemWorkflow.appendAction(
+      "attachInventoryItems",
+      CreateInventoryItemActions.createInventoryItems,
       {
-        manager: transactionManager,
-        inventoryService,
-        productVariantInventoryService,
-        productVariantService,
-      },
-      variant_id,
-      input
+        invoke: pipe(
+          {
+            invoke: {
+              from: CreateInventoryItemActions.createInventoryItems,
+              alias: "createdItems",
+            },
+          },
+          generateAttachInventoryToVariantHandler(
+            variant_id,
+            productVariantInventoryService
+          )
+        ),
+        compensate: pipe(
+          {
+            invoke: {
+              from: "attachInventoryItems",
+              alias: "attachedItems",
+            },
+          },
+          generateDetachInventoryItemFromVariantHandler(
+            productVariantInventoryService
+          )
+        ),
+      }
     )
+  }
+
+  const { result } = await createInventoryItemWorkflow.run({
```

Comment:
todo: I believe we should check if the workflow is enabled here before running it wdyt?

Reply:
Not sure I follow?

Reply:
Workflows are enabled through a feature flag

Reply:
Since this endpoint handler has always been through a workflow (it just hasn't been moved to @medusajs/workflows) I would argue that it's okay to not check for this. Additionally, these endpoints only work if you have @medusajs/inventory installed which should require workflows to interact with the core. 

Wdyt?

Reply:
Yes thats true, it is workflow to workflow, discard my comment 👌